classes/Settings/class.xudfSettingsFormGUI.php

```diff
@@ -118,7 +115,7 @@ protected function initForm(): void
         $input->addOption($opt);
         // only offer redirect to caller if referer contains a ref_id
         // since some proxy scenarios do not pass the complete referer
-        if (isset($_SERVER['HTTP_REFERER']) && str_contains($_SERVER['HTTP_REFERER'],'ref_id')) {
+        if (isset($_SERVER['HTTP_REFERER']) && str_contains($_SERVER['HTTP_REFERER'], 'ref_id')) {
```

Comment:
Pls use the ILIAS server wrapper `$this->http->request()->getServerParams()`

Apply the same pattern to all `$_` variables, if possible.